shims/common/src/main/java/org/apache/hadoop/hive/thrift/DefaultKrbCustomAuthenticationProviderImpl.java

```diff
@@ -0,0 +1,16 @@
+package org.apache.hadoop.hive.thrift;
+
+import javax.security.sasl.AuthenticationException;
+
+public class DefaultKrbCustomAuthenticationProviderImpl implements KrbCustomAuthenticationProvider {
+
+ /**
+  * This class will be called when you set
+  * hive.server2.authentication=KERBEROS and hive.server2.kerberos.custom.authentication.used = true
+  * with hive.server2.kerberos.custom.authentication.class=<class name> on hive-site.xml
```

Comment:
does this mean, "when hive.server2.kerberos.custom.authentication.class is unset"?